woocommerce/woocommerce-gutenberg-products-block
# Refactor cart shipping settings and shipping calculator

Updates the shipping calculator and shipping total row logic to match the designs and feedback in #1787. This turned out to be a complex refactor, but it does resolve a few issues.

Fixes #1835
Fixes #1787

This PR also:

- introduces context to handle sample cart data/rates avoiding the need to pass data down the component tree.
- (#1787) Makes the two shipping block options unrelated to each other so each option can be toggled individually.
- Tidies up the shipping calculator logic so it's clearer where addresses are updated
- Fixes some shapes/missing default data preventing cart notices (found this when implementing preview data).
- Allows `TotalsItem` to display regular text (required for #1787)
- Small update to `withFeedbackPrompt` to avoid it jumping around in position when toggling the block options on/off. Made the text consistent between the feedback prompt for cart and checkout to cut down i18n.

### Screenshots

No rates, with address:
![no rates found with address entered](https://user-images.githubusercontent.com/90977/76424890-c2108080-63a0-11ea-822f-0197b39f2714.png)

Rates found:
![with rates](https://user-images.githubusercontent.com/90977/76424894-c341ad80-63a0-11ea-9a63-c26c7aeb0a62.png)

No address entered yet, and no calculator enabled:
![no address no rates no calculator](https://user-images.githubusercontent.com/90977/76424897-c3da4400-63a0-11ea-9dda-1115991f0405.png)

No address entered yet, and no rates found:
![no address and no rates](https://user-images.githubusercontent.com/90977/76424898-c472da80-63a0-11ea-8ddc-8dcc4e62dd25.png)

### How to test the changes in this Pull Request:

1. Best to test in a private window. 
2. In wc settings, set the default address to "no location" so when you enter the cart page no address data will be set.
3. Ensure that when "Hide shipping costs until an address is entered" is enabled, rates only ever show after entering an address (country minimum)
4. Test that "Shipping calculator" option hides/shows the calculator respectively.
